### DISCUSSION TITLE
Browse Path Prefixed Integers: Are they assigned by NodeClass type?

### DISCUSSION THREAD

**Discussion by dldltn** [FreeOpcUa/python-opcua#1419] [Q&A] [answered]
Hey folks,

I'm trying to understand the Object/Variable structure of a server so that I can get node IDs by their browse path in a python function (which I'm already doing but I want to try to minimise issues with my code). 

I've noticed that the BrowseName of an Object is often in the following hierarchical structure: 

- 0:Objects
   - 2:Server
   - 2:Channel
      - 2:SomeObject
         - 2:Variable1
         - 2:Variable2
- 0:Types
   - 0:DataTypes
   - 0:EventTypes

And, I'm using the following code to get my node IDs:

`root = client.get_root_node()`
`node1 = root.get_child(["0:Objects", "2:Channel", "2:SomeObject", "2:Variable1"])`

Is it possible to do away with the prefixed integers altogether? For example:

`node1 = root.get_child("/Objects/Channel/SomeObject/Variable")`

If not, then how is the prefixed integer (i.e., the 0 or 2) assigned? Is it a matter of server setup?

~~I can't determine a relationship between the NodeClass (Object or Variable etc.) but I have noticed that the Node ID formats are different. For example, a Node ID of i=84 has its BrowseName prefixed with a "0:". Yet a Node ID of "ns=2;i=6" has a BrowseName prefixed with a "2:".~~

~~Do node classes have different types? If so, is it possible to get the node type? Then, I can just throw in an if statement based on the node type to handle it appropriately.~~

EDIT: I asked far too many questions here. I've struck through the _extra_ questions, and for now I'll read up more on namespaces.

**Comment by swamper123** [comment, marked as answer]
Hey ho,

this prefixed integers are the namespaceindex of your node. They are neccesarry to identify the correct namespace of your nodes.

https://documentation.unified-automation.com/uasdkcpp/1.7.4/html/L2UaAdrSpaceConceptNamespaces.html

**Reply by dldltn** [reply]
Thank you. I'll take the time to read the link you provided and learn more about namespaces.
